osp/core/utils/general.py

```diff
@@ -214,165 +295,255 @@ def get_custom_datatype_triples():
+    elif sum(bool(x) for x in (path, file)) > 1:
+        raise ValueError('Specify either a path or a file (a file-like object,'
+                         ' a string, or a json document dictionary) '
+                         'to read from.')
+    elif format not in ('json', 'xml', 'ttl'):
+        raise ValueError(f'Unsupported format {format}. The supported formats '
+                         f'are: {", ".join(("json", "xml", "ttl"))}. '
+                         f'Please check the docstring of this function for '
+                         f'more information.')
+
+    if path:
+        if not pathlib.Path(path).is_file():
+            raise Exception(f'{path} is not a file.')
+        with open(path, 'r') as file:
+            contents = file.read()
+            file_like = False
+    elif file:
+        if type(file) in (str, dict):
+            if type(file) is dict and format != 'json':
+                raise TypeError('Not a json document.')
+            contents = file
+            file_like = False
+        else:
+            if '__read__' not in file.__dir__():
+                raise TypeError(f'{file} is not a file-like object.')
+            contents = file
+            file_like = True
+
+    from osp.core.session.core_session import core_session
+    session = session or core_session
+    if format == 'json':
+        results = deserialize_cuds_object(contents, session=session,
+                                          buffer_context=None)
+    elif format in ('xml', 'ttl'):
+        results = import_rdf_file(contents if file_like
+                                  else io.StringIO(contents),
+                                  format=format, session=session,
+                                  buffer_context=None)
+    return results
+
+
+def export(item: Optional = None,
+           path: Optional[str] = None,
+           format: str = None,
+           rel: OntologyRelationship = cuba.activeRelationship,
+           max_depth: float = float("inf")) -> Union[str, None]:
+    """Exports CUDS objects in a variety of formats.
 
     Args:
-        path (str): Path to the rdf file to import.
-        format (str, optional): The file format of the file. Defaults to "xml".
-        session (Session, optional): The session to add the CUDS objects to.
-            Defaults to the CoreSession.
-        buffer_context (BufferContext, optional): Whether to add the objects
-            to the buffers of the user or the engine. Default is equivalent of
-            the user creating the CUDS objects by hand.. Defaults to None.
+        item (Union[Cuds, Session], optional): the CUDS object to serialize,
+            or a session to serialize all of its CUDS objects. If no item is
+            specified, then the coresession is exported.
+        path (str, optional): a path to save the CUDS objects. If no path is
+            specified, a string with the results will be
+            returned instead.
+        format(str): the target format.
```

Comment:
Support also mime-types, as supported in RDFLib.
Ref: https://rdflib.readthedocs.io/en/stable/plugin_serializers.html

Reply:
Done